cloudinit/net/networkd.py

```diff
@@ -271,7 +298,23 @@ def _render_content(self, ns):
                             name = dev_name
                             break
                 if name in ns.config["ethernets"]:
-                    self.dhcp_domain(ns.config["ethernets"][name], cfg)
+                    device = ns.config["ethernets"][name]
+                    for version in ["4", "6"]:
```

Comment:
This check is specific for VMware config. The following diff adds a comment underlying that and simplifying the if-condition. Feel free to rephrase it.

```diff
diff --git a/cloudinit/net/networkd.py b/cloudinit/net/networkd.py
index d9ad1fc8c..c54352956 100644
--- a/cloudinit/net/networkd.py
+++ b/cloudinit/net/networkd.py
@@ -299,12 +299,13 @@ class Renderer(renderer.Renderer):
                             break
                 if name in ns.config["ethernets"]:
                     device = ns.config["ethernets"][name]
+                    # dhcp{version}domain are extra keys only present in
+                    # VMware config.
                     for version in ["4", "6"]:
                         if (
                             f"dhcp{version}domain" in device
-                            and f"dhcp{version}-overrides" in device
                             and "use-domains"
-                            in device[f"dhcp{version}-overrides"]
+                            in device.get(f"dhcp{version}-overrides", {})
                         ):
                             exception = (
                                 f"{name} has both dhcp{version}domain"

```

Reply:
Thanks for the tip! I haven't written python in a while. Glad there was a better way :)